using java-dev images

arschles opened the pull request:
we now have `iron/java-dev` images that contain the JDK. will have new `iron/java` images (will open a PR in https://github.com/iron-io/dockers with those changes) that only contain the JRE (minus the GUI components) as well, which should be smaller than current.

cc/ @treeder

arschles committed:
using java-dev images

arschles committed:
what to use for compilation

treeder commented on java/README.md:
````diff
@@ -3,19 +3,19 @@
 
 ```sh
-docker run --rm -v "$(pwd)":/worker -w /worker iron/images:java-1.8 sh -c 'java -cp gson-2.2.4.jar:json-java.jar:ironworker.jar:. Worker101 -payload hello.payload.json -config hello.config.yml -id 123'
+docker run --rm -v "$(pwd)":/worker -w /worker iron/java:1.8 sh -c 'java -cp gson-2.2.4.jar:json-java.jar:ironworker.jar:. Worker101 -payload hello.payload.json -config hello.config.yml -id 123'
````
Should switch to the env vars usage like the rest of the examples.

arschles replied:
@treeder it's like this in `master` now. If I switch to env vars, the code throws a null pointer exception. Looks like the problem is on [line 24 of Worker101.java](https://github.com/iron-io/dockerworker/blob/master/java/Worker101.java#L24) since we pass cmdline args to `WorkerHelper.fromArgs`. Unless it's an easy fix or I have this all wrong, I'd rather save the code change for a later PR.

treeder replied:
Should update the java lib like you said, all our clients should support the ENV vars.

arschles replied:
updated to latest ironworker jar in https://github.com/iron-io/dockerworker/commit/17d3b8e42eea45b8cea715bdd0dd4a015cb7ce00. it falls back to env vars  like you suggested

arschles committed:
using env vars for payload file

and updating the ironworker jar to newest in maven central - it
respects env vars if you pass in an empty array

arschles committed:
fixing URL query

wikipedia redirects now to https and the previous code didn’t follow
redirs. making the request to the https URL would require building a
JVM trust store which makes things complicated, so using a simple URL
that doesn’t require https instead

arschles commented:
@treeder this is now ready

``` java
Aarons-MBP:java arschles$ docker run --rm -v "$(pwd)":/worker -w /worker iron/java-dev:1.8 sh -c 'javac -cp "json-java.jar:gson-2.2.4.jar:ironworker.jar" Worker101.java PayloadData.java'
Aarons-MBP:java arschles$ docker run --net=host -e PAYLOAD_FILE=hello.payload.json --rm -v "$(pwd)":/worker -w /worker iron/java:1.8 sh -c 'java -cp gson-2.2.4.jar:json-java.jar:ironworker.jar:. Worker101 -config hello.config.yml -id 123'
Running worker
http://httpbin.org/get?query=meerkats
{"query":"meerkats"}
```

treeder commented:
Can't merge.

arschles committed:
Merge branch 'master' into java-tiny-images

arschles commented:
@treeder all good now

treeder commented:
Can't merge again, another update please. ;)

arschles commented:
Will do this afternoon

Sent from my iPhone

> On Oct 2, 2015, at 11:41, Travis Reeder notifications@github.com wrote:
> 
> Can't merge again, another update please. ;)
> 
> —
> Reply to this email directly or view it on GitHub.

arschles committed:
Merge branch 'master' into java-tiny-images

arschles commented:
@treeder good to go
